src/irr/asset/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -13,43 +13,65 @@ void main()
     Normal = normalMat*normalize(vNormal);
 }
 )";
-    constexpr const char* FRAG_SHADER_NO_UV_MAIN =
-R"(#define Ia 0.1
-void main()
+    constexpr const char* FRAG_SHADER_NO_UV_BSDF_COS_EVAL =
+R"(#ifndef _IRR_BSDF_COS_EVAL_DEFINED_
+#define _IRR_BSDF_COS_EVAL_DEFINED_
+
+// Spectrum can be exchanged to a float for monochrome
+#define Spectrum vec3
+#define Ia 0.1
+
+//! This is the function that evaluates the BSDF for specific view and observer direction
+// params can be either BSDFIsotropicParams or BSDFAnisotropicParams 
+Spectrum irr_glsl_bsdf_cos_eval(in irr_glsl_BSDFIsotropicParams params)
 {
-    vec3 N = normalize(Normal);
-    vec3 L = normalize(-ViewPos);
-    vec3 R = -reflect(L,N);
-    float NdotL = max(dot(N,L), 0.0);
-    float VdotR = max(dot(L,R), 0.0);
+    vec3 Kd = PC.params.Kd;
 
-    vec3 color;
-    if (PC.params.extra&ILLUM_MODEL_MASK > 0)
+    vec3 color = vec3(0.0);
+    if ((PC.params.extra&ILLUM_MODEL_MASK) > 0)
     {
-        color = PC.params.Ka*Ia + PC.params.Kd*NdotL;
+        vec3 Ka = PC.params.Ka;
+        vec3 Ks = PC.params.Ks;
+        float Ns = PC.params.Ns;
+
+        vec3 diff = Kd*params.NdotL;
         switch (PC.params.extra&ILLUM_MODEL_MASK)
         {
+        case 1:
+            color = Ia*Ka + diff;
+            break;
         case 2:
-        case 3://2 + reflection map
-        case 4://3 with transparency (glass)
-        case 6:
-        case 8://reflection map
-        case 9://reflection map
-            color += PC.params.Ks*pow(VdotR, PC.params.Ns);
+        case 3://2 + IBL
+        case 5://basically same as 3
+        case 8://basically same as 5
+        {
+            float fr = Fresnel_dielectric(PC.params.Ni, params.NdotL);
+            vec3 spec = Ks*pow(params.NdotH, Ns);
+            color += Ia*Ka + mix(diff, spec, fr);
+        }
             break;
-        case 5:
+        case 4:
+        case 6:
         case 7:
-            color += PC.params.Ks*pow(VdotR, PC.params.Ns)*Fresnel_dielectric(PC.params.Ni, NdotL);
+        case 9://basically same as 4
+        {
+            float fr = Fresnel_dielectric(PC.params.Ni, params.NdotL);
+            vec3 spec = Ks*pow(params.NdotH, Ns);
+            color = fr*(Ia*Ka+spec);
```

Comment:
The "ambient" is not a extra light factor added to the current light with its own direction, ambient is an area light coming from all directions, hence ambient is unaffected by fresnel (if you assume fresnel passes white furnace, which it should).